src/vs/workbench/contrib/search/browser/searchActions.ts

```diff
@@ -585,28 +585,34 @@ export class RemoveAction extends AbstractSearchAndReplaceAction {
 		super(Constants.RemoveActionId, appendKeyBindingLabel(RemoveAction.LABEL, keyBindingService.lookupKeybinding(Constants.RemoveActionId), keyBindingService), ThemeIcon.asClassName(searchRemoveIcon));
 	}
 
-	override run(): Promise<any> {
+	override async run(): Promise<any> {
 		const opInfo = getElementsToOperateOnInfo(this.viewer, this.element, this.configurationService.getValue<ISearchConfigurationProperties>('search'));
 		const elementsToRemove = opInfo.elements;
 
-		const currentBottomFocusElement = elementsToRemove[elementsToRemove.length - 1];
-
-		const nextFocusElement = opInfo.mustReselect && (!currentBottomFocusElement || currentBottomFocusElement instanceof SearchResult || arrayContainsElementOrParent(currentBottomFocusElement, elementsToRemove)) ?
-			this.getElementToFocusAfterRemoved(this.viewer, <any>currentBottomFocusElement) :
-			null;
+		if (elementsToRemove.length === 0) {
+			return;
+		}
 
-		if (nextFocusElement) {
-			this.viewer.reveal(nextFocusElement);
-			this.viewer.setFocus([nextFocusElement], getSelectionKeyboardEvent());
-			this.viewer.setSelection([nextFocusElement], getSelectionKeyboardEvent());
+		if (opInfo.mustReselect) {
+			for (const currentElement of elementsToRemove) {
+				const nextFocusElement = !currentElement || currentElement instanceof SearchResult || arrayContainsElementOrParent(currentElement, elementsToRemove) ?
+					this.getElementToFocusAfterRemoved(this.viewer, currentElement) :
+					null;
+				if (nextFocusElement && !elementsToRemove.includes(nextFocusElement)) {
+					this.viewer.reveal(nextFocusElement);
```

Comment:
Seems that there's no repro for replace although folderMatches and fileMatches use `getElementToFocusAfterRemoved` and this is because it doesn't call `this.viewer.reveal` in that case (so no brokenness, just maybe focusing wrong at first). 

This probably also doesn't focus 100% correctly since fileMatches opt to navigate to the next filematch (and same with matches and foldermatches). I think there needs to be a bigger conversation about what should focus when something is dismissed or replaced because it seems that most removal/replace is trying to navigate to the next tree item of the same level, which doesn't work when you can multi-select elements of multiple types. Should we be always trying to focus next on the next match? or any next element that's valid? I think that there might need to be a lot of refactoring behind that and I'm not too sure if it should be in this PR or a different one.